.gitignore

```diff
@@ -21,3 +20,135 @@ image_files*
 
 # datasets
 cifar*
+
+# Byte-compiled / optimized / DLL files
+*.py[cod]
+*$py.class
+.direnv
+.idea
+.idea/**
+.vscode
+.vscode/**
+
+# C extensions
+*.so
+**.txt
```

Comment:
why are you proposing to add: **.txt ?
That seems odd to ignore to me

Reply:
I copied them from my other repos, but I can remove some of them if you want. I can delete a bunch of other things that are more related to personal choices.

Reply:
making some changes based on personal preferences is fine! But it will be helpful to clarify which additions & deletions are based on personal preferences VS which additions & deletions are based on improving the code organization